openpype/pipeline/farm/patterning.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+import re
+
+
+def match_aov_pattern(app, aov_patterns, render_file_name):
+    """Matching against a `AOV` pattern in the render files.
+
+    In order to match the AOV name we must compare
+    against the render filename string that we are
+    grabbing the render filename string  from the collection
+    that we have grabbed from `exp_files`.
+
+    Args:
+        app (str): Host name.
+        aov_patterns (list): List of AOV patterns from AOV filters.
+        render_file_name (str): Incoming file name to match against.
+
+    Returns:
+        bool: Review state for rendered file (render_file_name).
+    """
+    aov_pattern = aov_patterns.get(app, [])
+    if aov_pattern:
+        if re.match(aov_pattern, render_file_name):
+            preview = True
+            return preview
+        else:
+            return False
```

Comment:
I think that @BigRoy  wanted to say that you don't need those conditions at all because the can be replaced completely with the result of `match`. There is also unhandled situation when the pattern is not set.
```suggestion
    if not aov_pattern:
        return False
    return re.match(aov_pattern, render_file_name)
```

Reply:
Thank you @iLLiCiTiT, I'll fix the unhandled situation for an unset pattern, but for `re.match` my understanding is it returns `None` if it doesn't match.  I might be misunderstanding the docs here: https://docs.python.org/3/library/re.html#re.match 

Please let me know so I can fix this if I'm wrong.

Reply:
You're right. It can still be simplified (Edited the suggestion). Question is how to handle "empty regex string". Does it match in that case, or it should return `False` like in the suggestion?

Reply:
Thank you.  Based on my understanding it should return `False ` so it sets the review status (and that was the intended behavior at the beginning).  If a studio (or artist) wants to, they'd need to specify the regex, imho I think assuming intended behavior needing to be explicit would always be a good idea.

Reply:
I wonder if it's worth it to use  `re.fullmatch` (https://docs.python.org/3/library/re.html#re.fullmatch) so as to simplify mainly adding to the benefit of being explicit about it, like such:

```
if re.fullmatch(aov_pattern, render_file_name):
    return True
else:
    return False

```